go/libraries/doltcore/sqle/database.go

```diff
@@ -657,7 +777,18 @@ func (db Database) Flush(ctx *sql.Context) error {
 		return err
 	}
 
-	return db.SetRoot(ctx, newRoot)
+	err = db.SetRoot(ctx, newRoot)
+	if err != nil {
+		return nil
+	}
+
+	// Flush any changes made to temporary tables
+	newTempTableRoot, err := dsess.tempTableEditSessions[db.Name()].Flush(ctx)
+	if err != nil {
+		return nil
+	}
+
+	return dsess.SetTempTableRoot(ctx, db.Name(), newTempTableRoot)
```

Comment:
This is a problem... we need to update either both roots or neither of them, atomically

I think a TODO is fine for now